python/cuml/dask/ensemble/randomforestclassifier.py

```diff
@@ -357,16 +375,47 @@ def predict(self, X, output_class=True, algo='auto', threshold=0.5,
             preds = self.predict_model_on_cpu(X=X,
                                               convert_dtype=convert_dtype)
         else:
-            preds = \
-                self._predict_using_fil(X, output_class=output_class,
-                                        algo=algo,
-                                        threshold=threshold,
-                                        convert_dtype=convert_dtype,
-                                        fil_sparse_format=fil_sparse_format,
-                                        delayed=delayed)
-
+            if broadcast_data:
+                preds = \
+                    self.partial_inference(
+                        X,
+                        algo=algo,
+                        convert_dtype=convert_dtype,
+                        fil_sparse_format=fil_sparse_format,
+                        delayed=delayed
+                    )
+            else:
+                preds = \
+                    self._predict_using_fil(
+                        X,
+                        output_class=output_class,
+                        algo=algo,
+                        threshold=threshold,
+                        convert_dtype=convert_dtype,
+                        fil_sparse_format=fil_sparse_format,
+                        delayed=delayed
+                    )
         return preds
 
+    def partial_inference(self, X, delayed, **kwargs):
+        partial_infs = \
+            self._partial_inference(X=X,
+                                    op_type='classification',
+                                    delayed=delayed,
+                                    **kwargs)
+
+        def reduce(partial_infs, n_estimators_per_worker, unique_classes):
+            votes = dask.array.average(partial_infs, axis=1,
+                                       weights=n_estimators_per_worker)
+            votes = votes.compute()
+            votes = votes.argmax(axis=1)
```

Comment:
Nitpick here, but could we use a different variable name for the final assignment? After computing votes.argmax(..), unless I'm not understanding, this should be something like 'pred_class_index' or something. As written its a little confusing as to what is being returned.